### DISCUSSION TITLE
[Idea] Detailed information about cart rule apllication in an order

### DISCUSSION THREAD

**Discussion by Javier23408** [PrestaShop/PrestaShop#35963] [Ideas]
### The problem

When it is done a order, it is unknown the price of a cart_rule applied on an order line (order_detail). 

Right now we only know the total of the cart rule applied to the entire order (in ps_order_cart_rule table) without knowing the details.

In some cases you want to know what part of the total cart_rule is applied to the product.

### The solution

Create new columns in ps_order_detail table or new ps_order_detail_cart_rule table to record the proportional part applied to each product or line.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by Hlavtox** [comment]
@kpodemski Is it even possible to apply a cart rule to order line? I don't think so.
@Javier23408 Aren't you talking about a specific price?

**Reply by Javier23408** [reply]
Yes, it is possible to apply a cart rule to a specific product or some products in a specific category, or brand, or... etc.

The 3 conditions are: cart rule applied in carrier, applied in customer groups (all products in an order) or in products.

**Reply by Hlavtox** [reply]
Yes, but prestashop just adds a discount with some amount and finished. It doesn't "remember" what was the product it applied the discount to.

![Snímek obrazovky 2024-04-23 134135](https://github.com/PrestaShop/PrestaShop/assets/6097524/26d981b6-8ed6-47a5-a756-08420e180a15)

**Reply by Javier23408** [reply]
I understand, my idea is that I want Prestashop to remember that a proportional part of the discount is applied to each line. Because the discount can be applied to some products and not to others within the order.

Now we only have the total discount on the order as you say, but I want to add more complete and detailed information about how this discount is obtained.

That's why I propose to create a table ps_order_detail_cart_rule, or add columns in ps_order_detail.

**Reply by kpodemski** [reply]
@Javier23408 thing is that PrestaShop doesn't really put the discount on the product line, but on the cart itself, and if you want to change it, it requires more effort
